app/ante/ante.go

```diff
@@ -1,67 +1,89 @@
 package ante
 
 import (
+	"fmt"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keys/ed25519"
+	"github.com/cosmos/cosmos-sdk/crypto/keys/secp256k1"
+	"github.com/cosmos/cosmos-sdk/crypto/types/multisig"
 	sdk "github.com/cosmos/cosmos-sdk/types"
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	"github.com/cosmos/cosmos-sdk/x/auth"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
 	authante "github.com/cosmos/cosmos-sdk/x/auth/ante"
-	"github.com/cosmos/cosmos-sdk/x/auth/keeper"
-	"github.com/cosmos/cosmos-sdk/x/auth/types"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtypes "github.com/cosmos/cosmos-sdk/x/auth/types"
 
 	"github.com/cosmos/ethermint/crypto/ethsecp256k1"
 	evmtypes "github.com/cosmos/ethermint/x/evm/types"
-
-	tmcrypto "github.com/tendermint/tendermint/crypto"
 )
 
-func init() {
-	ethsecp256k1.RegisterCodec(types.ModuleCdc)
-}
-
 const (
 	// TODO: Use this cost per byte through parameter or overriding NewConsumeGasForTxSizeDecorator
 	// which currently defaults at 10, if intended
 	// memoCostPerByte     sdk.Gas = 3
 	secp256k1VerifyCost uint64 = 21000
 )
 
+// AccountKeeper defines an expected keeper interface for the auth module's AccountKeeper
+type AccountKeeper interface {
+	authante.AccountKeeper
+	NewAccountWithAddress(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI
+	GetAccount(ctx sdk.Context, addr sdk.AccAddress) authtypes.AccountI
+	SetAccount(ctx sdk.Context, account authtypes.AccountI)
+}
+
+// BankKeeper defines an expected keeper interface for the bank module's Keeper
+type BankKeeper interface {
+	authtypes.BankKeeper
+	GetBalance(ctx sdk.Context, addr sdk.AccAddress, denom string) sdk.Coin
+	SetBalance(ctx sdk.Context, addr sdk.AccAddress, balance sdk.Coin) error
+}
+
 // NewAnteHandler returns an ante handler responsible for attempting to route an
 // Ethereum or SDK transaction to an internal ante handler for performing
 // transaction-level processing (e.g. fee payment, signature verification) before
 // being passed onto it's respective handler.
-func NewAnteHandler(ak auth.AccountKeeper, evmKeeper EVMKeeper, sk types.SupplyKeeper) sdk.AnteHandler {
+func NewAnteHandler(
+	ak AccountKeeper, bankKeeper BankKeeper, evmKeeper EVMKeeper, signModeHandler authsigning.SignModeHandler,
+) sdk.AnteHandler {
 	return func(
 		ctx sdk.Context, tx sdk.Tx, sim bool,
 	) (newCtx sdk.Context, err error) {
 		var anteHandler sdk.AnteHandler
 		switch tx.(type) {
-		case auth.StdTx:
+		case *evmtypes.MsgEthereumTx:
+			anteHandler = sdk.ChainAnteDecorators(
+				NewEthSetupContextDecorator(), // outermost AnteDecorator. EthSetUpContext must be called first
+				authante.NewRejectExtensionOptionsDecorator(),
+				NewEthMempoolFeeDecorator(evmKeeper),
+				authante.NewValidateBasicDecorator(),
```

Comment:
i may be totally wrong here but looks like the SDK definition of `ValidateBaic()` is to simply return nil without any proper checks (like how the StdTx's ValidateBasic method had done).

Reply:
it just checks that the tx field validation passes